### ISSUE TITLE
Problem when predicting pose using shared Resnet

### ISSUE THREAD

**Issue by biggiantpigeon** [nianticlabs/monodepth2#374]
I have been working fine with separated resnet when predicting pose, but things gets weird when I use shared resnet: I got almost same result when swap two input images. Where did I do wrong?
so this is my related code:

```
#network define
encoder = networks.ResnetEncoder(18, False).cuda()
if pose_net_type == "separate":
    pose_encoder = networks.ResnetEncoder(18, False, 2).cuda()
    pose_decoder = networks.PoseDecoder(pose_encoder.num_ch_enc, 1, 2).cuda()
elif pose_net_type == "shared":
    #pose_encoder = encoder
    pose_decoder = networks.PoseDecoder(pose_encoder.num_ch_enc, 2).cuda()

#predict pose
if pose_net_type == "separate":
    features_pose = [pose_encoder(torch.cat([prev_image_pytorch, input_image_pytorch], 1))]
elif pose_net_type == "shared":
    feature_1 = encoder(prev_image_pytorch)
    feature_2 = encoder(input_image_pytorch)
    features_pose = [feature_1, feature_2]
axisangle, translation = pose_decoder(features_pose)
```
I swap two images by doing
`	features_pose = [feature_2, feature_1]`

This is my result of output from function transform_from_parameters:
**when using separate resnet:**
image1 -> image2
![image](https://user-images.githubusercontent.com/80443901/123926655-adedf980-d9be-11eb-935f-14a32e28629e.png)
image2 -> image1
![image](https://user-images.githubusercontent.com/80443901/123926689-b7776180-d9be-11eb-8d66-7bbabf664249.png)
which is obviously a transform pair.

**when using shared resnet:**
image1 -> image2
![image](https://user-images.githubusercontent.com/80443901/123926806-d970e400-d9be-11eb-8ec1-a964c60d8221.png)
image2 -> image1
![image](https://user-images.githubusercontent.com/80443901/123926895-eb528700-d9be-11eb-91c4-41cf07da100a.png)
which are almost the same when I swap features.
By the way, when I run pose prediction of a image sequence to get the trajectory, both result(pred in the figure) seem to fit well with my groundtruth(GT in the figure), they look almost the same, but different.
![image](https://user-images.githubusercontent.com/80443901/123929060-0cb47280-d9c1-11eb-9a03-aa1c6ceed41b.png)

I trained the model with about 8k images, 27 epochs, the loss did go down, but very slow and not very much(from 0.1 to 0.06), I don't know if this is enough.
![image](https://user-images.githubusercontent.com/80443901/123928939-edb5e080-d9c0-11eb-85ad-f87d1c399bc4.png)

Thanks in advance!
